### ISSUE TITLE
Cannot calculate loss for each data

### ISSUE THREAD

**Issue by unnonouno** [chainer/chainer#1574]
All loss functions calculates **average** of loss of mini-batch. We cannot calculates loss of each data nor sum of loss.
I think `F.average(F.softmax_cross_entropy(x, t))` and `F.sum(F.softmax_cross_entropy(x, t))` are more useful and efficient enough.

**Comment by delta2323** [comment]
I once tried to implement a PR (#1223) that changes the behavior of the loss function. It includes your suggestion. But as it is halfway and slightly complicated, I would like to discuss the design.

**Comment by unnonouno** [comment]
I need this feature, and I made a elementwise version of loss function by myself and use it in my project.
I feel separating the current implementation with elementwise loss and aggregation function is enough.

**Comment by unnonouno** [comment]
solved via #2558
